pkg/agent/cniserver/pod_configuration.go

```diff
@@ -430,59 +419,33 @@ func (pc *podConfigurator) setupContainerOVSPort(
 	}
 }
 
-func removeContainerLink(containerID string, containerNetns string, ifname string) error {
-	if err := ns.WithNetNSPath(containerNetns, func(_ ns.NetNS) error {
-		var err error
-		_, err = ip.DelLinkByNameAddr(ifname)
-		if err != nil && err == ip.ErrLinkNotFound {
-			// Not found link should return success for deletion
-			klog.V(2).Infof("Interface %s not found in netns %s", ifname, containerNetns)
-			return nil
-		}
-		return err
-	}); err != nil {
-		klog.Errorf("Failed to delete interface %s of container %s: %v", ifname, containerID, err)
-		return err
-	}
-	return nil
-}
-
-func (pc *podConfigurator) removeInterfaces(podName, podNamespace, containerID, containerNetns, ifname string) error {
-	if containerNetns != "" {
-		if err := removeContainerLink(containerID, containerNetns, ifname); err != nil {
-			return err
-		}
-	} else {
-		// From the CNI spec for the DEL command:
-		// When CNI_NETNS and/or prevResult are not provided, the plugin should clean up as
-		// many resources as possible (e.g. releasing IPAM allocations) and return a
-		// successful response.
-		// In our case it means deleting the OVS port.
-		klog.V(2).Infof("Target netns not specified, not removing veth pair")
-	}
-
+func (pc *podConfigurator) removeInterfaces(podName, podNamespace, containerID string) error {
 	containerConfig, found := pc.ifaceStore.GetContainerInterface(podName, podNamespace)
 	if !found {
 		klog.V(2).Infof("Did not find the port for container %s in local cache", containerID)
 		return nil
 	}
 
-	portUUID := containerConfig.PortUUID
-	ovsPortName := containerConfig.InterfaceName
-	klog.V(2).Infof("Deleting OVS port with UUID %s peer container %s", portUUID, containerID)
-	// Remove Openflow entries of target container
-	if err := pc.ofClient.UninstallPodFlows(ovsPortName); err != nil {
-		klog.Errorf("Failed to delete Openflow entries for container %s: %v", containerID, err)
-		return err
+	klog.V(2).Infof("Deleting veth devices for container %s", containerID)
+	// Don't return an error if the device is already removed as CniDel can be called multiple times.
+	err := ip.DelLinkByName(containerConfig.InterfaceName)
+	if err != nil && err != ip.ErrLinkNotFound {
```

Comment:
is it worth adding a log message for the `ErrLinkNotFound` case?

Reply:
Done